src/main.c

```diff
@@ -496,56 +496,99 @@ void eggAssert(const char *file, int line, const char *module)
+  }
+  printf("Configure flags: " EGG_AC_ARGS "\n");
+  printf("Compiled with: ");
+#ifdef IPV6
+  printf("IPv6, ");
+#endif
+#ifdef TLS
+  printf("TLS, ");
+#endif
+  printf("handlen=%d\n", HANDLEN);
+  bg_send_quit(BG_ABORT);
+}
+
+/* Hard coded text because config file isn't loaded yet,
+   meaning other languages can't be loaded yet.
+   English (or an error) is the only possible option.
+*/
+void show_help() {
+  printf("\n%s\n\n", version);
+  printf("Usage: eggdrop [options] [config-file]\n\n"
+         "Options:\n"
+         "-n Don't background; send all log entries to console.\n"
+         "-nc  Don't background; display channel stats every 10 seconds.\n"
+         "-nt  Don't background; use terminal to simulate DCC chat.\n"
+         "-m   Create userfile.\n"
+         "-h   Show this help.\n"
+         "-v   Show version info, then quit.\n\n");
+  bg_send_quit(BG_ABORT);
+}
+
+static void do_arg()
+{
+  int option = 0;
+/* Bitmask structure to hold cli flags
+   | QUIT| BAD FLAG| h| n| c| t| m| v|
+   |  128|       64|32|16| 8| 4| 2| 1|
+*/
+  unsigned char cliflags = 0;
+
+  while ((option = getopt(argc, argv, "hnctmv")) != -1) {
+    switch (option) {
       case 'n':
+        cliflags |= 16;
         backgrd = 0;
         break;
       case 'c':
+        cliflags |= 8;
         con_chan = 1;
         term_z = 0;
         break;
       case 't':
+        cliflags |= 4;
         con_chan = 0;
         term_z = 1;
         break;
       case 'm':
+        cliflags |= 2;
         make_userfile = 1;
         break;
       case 'v':
-        strncpyz(x, egg_version, sizeof x);
-        newsplit(&z);
-        newsplit(&z);
-        printf("%s\n", version);
-        if (z[0])
-          printf("  (patches: %s)\n", z);
-        printf("Configure flags: " EGG_AC_ARGS "\n");
-        printf("Compiled with: ");
-#ifdef IPV6
-        printf("IPv6, ");
-#endif
-#ifdef TLS
-        printf("TLS, ");
-#endif
-        printf("handlen=%d\n", HANDLEN);
-        bg_send_quit(BG_ABORT);
-        exit(0);
-        break;                  /* this should never be reached */
+        cliflags |= 129;		//128 + 1
+        break;
       case 'h':
-        printf("\n%s\n\n", version);
-        printf("%s\n", EGG_USAGE);
-        bg_send_quit(BG_ABORT);
-        exit(0);
-        break;                  /* this should never be reached */
-      }
-    } else
-    strncpyz(configfile, s, sizeof configfile);
+        cliflags |= 160;		//128 + 32
+        break;
+      default:
+        cliflags |= 192;		//128 + 64
+        break;
+    }
+  }
+  if ((cliflags & 64) || (cliflags & 32)) {
+    show_help();
+    exit(0);
+  } else if (cliflags & 1) {
+    show_ver();
+    exit(0);
+  } else {
+    if (argc > (optind + 1)) {
+      printf("\n");
+      printf("WARNING: More than one config file value detected\n");
+      printf("         Using %s as config file\n", argv[optind]);
+    }
+    strncpyz(configfile, argv[optind], sizeof configfile);
```

Comment:
Should still work based on ```if (argc > (optind + 1)) {``` ... ie, if the total number of args given is than 2 more than the number of flags

```
$ ./eggdrop -m patchtest.conf 

Eggdrop v1.8.1+alpha (C) 1997 Robey Pointer (C) 1999-2017 Eggheads
[09:44:15] --- Loading eggdrop v1.8.1+alpha (Tue Apr 25 2017)
[09:44:15] Listening at telnet port 3183 (all).
[13:44:15] Module loaded: dns             
[13:44:15] Module loaded: channels        
[13:44:15] Module loaded: server          
[13:44:15] Module loaded: ctcp            
[13:44:15] Module loaded: irc             
[13:44:15] Module loaded: transfer         (with lang support)
[13:44:15] Module loaded: share           
[13:44:15] Module loaded: notes            (with lang support)
[13:44:15] Module loaded: console          (with lang support)
[13:44:15] Module loaded: blowfish        
[13:44:15] Writing channel file...
[13:44:15] Userfile loaded, unpacking...
USERFILE ALREADY EXISTS (drop the '-m')
[13:44:15] === scooby: 0 channels, 130 users.
Launched into the background  (pid: 64929)
```